privacyidea.py

```diff
@@ -2,131 +2,241 @@
 from org.gluu.oxauth.service import AuthenticationService, SessionIdService
 from org.gluu.model.custom.script.type.auth import PersonAuthenticationType
 from org.gluu.service.cdi.util import CdiUtil
+from org.gluu.oxauth.service import UserService
+from org.gluu.util import StringHelper
 
+from javax.faces.application import FacesMessage
+from org.gluu.jsf2.message import FacesMessages
 
+from java.util import Arrays
 
-# TODO: Import the privacyIDEA Java SDK
 
-import java
+def logFromSDK(message):
+    print("privacyIDEA. JavaSDK: " + message)
 
 
 class PersonAuthentication(PersonAuthenticationType):
+
     def __init__(self, currentTimeMillis):
+        print("__init__")
         self.currentTimeMillis = currentTimeMillis
-        self.identity = CdiUtil.bean(Identity)
-        self.sessionIdService = CdiUtil.bean(SessionIdService)
-        self.authenticationService = CdiUtil.bean(AuthenticationService)
 
     def init(self, configurationAttributes):
-        print "privacyIDEA. Initialization"
+        print("privacyIDEA. init")
+
+        sdk_path = "/opt/java_sdk.jar"
+        if configurationAttributes.contains("sdk_path"):
+            sdk_path = configurationAttributes.get("sdk_path").getValue2()
+
+        import sys
+        sys.path.append(sdk_path)
+        try:
+            from org.privacyidea import Challenge
```

Comment:
Can we also have these imports at the beginning? Or will the scripts fail to load and gluu not run at all?

Reply:
If the script fails to load, the login is impossible. Since the sdk_path is configurable, it has to be loaded first before it is possible to import. Moreover, with the sdk_path known, an error message can be logged to indicate where the sdk should be.

Reply:
My bad. I understand, this imports the java library ``Challenge`` and it can of course not be loaded without the path. Thanks.